File: app/src/main/java/vergecurrency/vergewallet/viewmodel/ServiceURLViewModel.java
package vergecurrency.vergewallet.viewmodel;

import androidx.lifecycle.ViewModel;

import vergecurrency.vergewallet.Constants;
import vergecurrency.vergewallet.service.model.PreferencesManager;

public class ServiceURLViewModel extends ViewModel {

	public String getCurrentServiceURL() {
		return PreferencesManager.getWalletServiceUrl();
	}

	public void setNewServiceURL(String serviceURL) {
		PreferencesManager.setWalletServiceUrl(serviceURL);
	}

	public void setDefaultServiceURL(){
		setNewServiceURL(Constants.VWS_ENDPOINT);
	}
}

File: app/src/main/kotlin/vergecurrency/vergewallet/viewmodel/ServiceURLViewModel.kt
package vergecurrency.vergewallet.viewmodel

import androidx.lifecycle.ViewModel

import vergecurrency.vergewallet.Constants
import vergecurrency.vergewallet.service.model.PreferencesManager

class ServiceURLViewModel : ViewModel() {

    val currentServiceURL: String
        get() = PreferencesManager.walletServiceUrl!!

    fun setNewServiceURL(serviceURL: String) {
        PreferencesManager.walletServiceUrl = serviceURL
    }

    fun setDefaultServiceURL() {
        setNewServiceURL(Constants.VWS_ENDPOINT)
    }
}
